Feature/26/docker integration

YMikevich opened the pull request.

YMikevich committed:
#24 feign client implemented, findAllAccountsById() method implemented
in spring-data-jpa-user-service

YMikevich committed:
#24 code refactored

YMikevich committed:
#24 code refactored

YMikevich committed:
#24 code refactored

YMikevich committed:
#25 basic routing in api gateway implemented

YMikevich committed:
Merge branch 'dev' into feature/26/docker-integration

YMikevich committed:
#26 spring cloud microservices wrapped up in docker

akraskovski commented on api-gateway/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+RUN apk update && apk add bash
+
+VOLUME /tmp
+ADD ./target/api-gateway-0.0.1-SNAPSHOT.jar api-gateway.jar
```
let's unlink in all dockerfiles from specific versions and replace them by regex search pattern.

akraskovski commented on api-gateway/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+VOLUME /tmp
+ADD ./target/api-gateway-0.0.1-SNAPSHOT.jar api-gateway.jar
+EXPOSE 8762
+ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-jar","/api-gateway.jar"]
```
are you sure that `"-Djava.security.egd=file:/dev/./urandom"` is necessary argument?

akraskovski commented on clients/es-integration/build.gradle:
```diff
@@ -34,10 +34,13 @@ dependencies {
     implementation 'org.springframework.cloud:spring-cloud-starter-config'
     implementation 'org.springframework.cloud:spring-cloud-starter-netflix-eureka-client'
+    implementation 'org.springframework.retry:spring-retry:1.2.4.RELEASE'
+    implementation 'org.springframework.boot:spring-boot-starter-aop:2.1.8.RELEASE'
```
do you need to specify dependencies versions or it's already exists in spring-boot-dependencies?

akraskovski commented on docker-compose.yml:
```diff
@@ -69,6 +95,34 @@ services:
+    build:
+      context: ./api-gateway
+    ports:
+      - '8762:8762'
```
what sense of your api-gateway if you're producing all ports from each server to host machine?
the idea was to hide all services except api-gateway inside the docker process and provide only one port to end server.

YMikevich committed:
#26 code refactored: useless dependencies deleted, regular expressions
added in Dockerfiles to match different versions of jars, timeouts
configured in api-gateway
